rules/private/diff_test_private.bzl

```diff
@@ -0,0 +1,170 @@
+# Copyright 2019 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Implementation of diff_test macro and underlying rule.
+
+The rule compares two binary files at build time, then writes a simple script
+that is the actual test executable. The script's exit code matches the result of
+the file comparison: zero if the files are equal and were expected to be so (or
+are different and were expected to be so), non-zero otherwise.
+"""
+
+def _impl(ctx):
+    if ctx.attr.is_windows:
```

Comment:
It may be more readable to have two functions, one for Windows, one for Linux (to avoid a long if block).

Reply:
Done.